colorbleed/plugins/maya/load/load_yeti_cache.py

```diff
@@ -46,70 +46,16 @@ def load(self, context, name=None, namespace=None, data=None):
 
         # Get node name from JSON
-        nodes = []
-        for node, settings in fursettings.items():
-
-            # Create transform
-            transform_name = "{}:{}".format(namespace, node.split("Shape")[0])
-            transform_node = cmds.createNode("transform", name=transform_name)
-
-            # Create new pgYetiMaya node
-            node_name = "{}:{}".format(namespace, node)
-            yeti_node = cmds.createNode("pgYetiMaya",
-                                        name=node_name,
-                                        parent=transform_node)
-
-            cmds.connectAttr("time1.outTime", "%s.currentTime" % yeti_node)
-
-            # Apply explicit colorbleed ID to node
-            shape_id = settings["cbId"]
-            asset_id = shape_id.split(":", 1)[0]
-
-            lib.set_id(node=yeti_node,
-                       unique_id=shape_id,
-                       overwrite=True)
-            settings.pop("cbId", None)
-
-            # Apply new colorbleed ID to transform node
-            # TODO: get ID from transform in data to ensure consistency
-            _ids = lib.generate_ids(nodes=[transform_node], asset_id=asset_id)
-            for n, _id in _ids:
-                lib.set_id(n, unique_id=_id)
-
-            # Apply settings
-            for attr, value in settings.items():
-                attribute = "%s.%s" % (yeti_node, attr)
-                cmds.setAttr(attribute, value)
-
-            # Ensure the node has no namespace identifiers
-            node = node.replace(":", "_")
-
-            # Create full cache path
-            cache = os.path.join(self.fname, "{}.%04d.fur".format(node))
-            cache = os.path.normpath(cache)
-            cache_fname = self.validate_cache(cache)
-            cache_path = os.path.join(self.fname, cache_fname)
-
-            # Preset the viewport density
-            cmds.setAttr("%s.viewportDensity" % yeti_node, 0.1)
-
-            # Add filename to `cacheFileName` attribute
-            cmds.setAttr("%s.cacheFileName" % yeti_node,
-                         cache_path,
-                         type="string")
-
-            cmds.setAttr("%s.imageSearchPath" % yeti_node,
-                         image_search_path,
-                         type="string")
-
-            # Set verbosity for debug purposes
-            cmds.setAttr("%s.verbosity" % yeti_node, 2)
-
-            # Enable the cache by setting the file mode
-            cmds.setAttr("%s.fileMode" % yeti_node, 1)
-
-            nodes.append(yeti_node)
-            nodes.append(transform_node)
+        if "nodes" in fursettings:
+            node_data = fursettings["nodes"]
+            nodes = self.create_nodes(namespace, image_search_path, node_data)
+        else:
+            # Backwards compatibilty
+            self.log.info("Encountered old data, "
+                          "using backwards compatibility")
+            nodes = self.create_nodes_old(namespace,
```

Comment:
Blegh, I so hate this. backwards compatibility; blegh!

Anyhow, would this make more sense to "update" the older data to the new one so we could pass it along?

Say you'd find the old data you'd refactor it to match to the new data and apply that instead? That way you wouldn't need the duplicated "old/new" code.

Reply:
Currently the cleaning of the published files is possible and possibly the road of least resistance.
I will refactor the code AFTER I know for sure the updating is correct.